src/main/kotlin/mate/academy/Main.kt

```diff
@@ -1,5 +1,78 @@
 package mate.academy
 
+import mate.academy.model.Color
+import mate.academy.service.FigureSupplier
+
 fun main() {
     // draw all figures here
+    val figureSupplier = FigureSupplier()
+
+    val figures: Array<Figure> = Array(6) {
+        if (it < 3) {
+            figureSupplier.getRandomFigure()
+        } else {
+            figureSupplier.getDefaultFigure()
+        }
+    }
+
+    for (figure in figures) {
+        figure.draw()
+    }
+}
+
+interface Figure {
+    var color: Color
+    fun getArea(): Double
+    fun draw()
+}
+
+class Square(val sideLength: Double, override var color: Color) : Figure {
+    override fun getArea(): Double {
+        return sideLength * sideLength
+    }
+
+    override fun draw() {
+        println("Square with side length: $sideLength, color: $color, area: ${getArea()}")
+    }
+}
+
+class Rectangle(val width: Double, val length: Double, override var color: Color) : Figure {
+    override fun getArea(): Double {
+        return width * length
+    }
+
+    override fun draw() {
+        println("Rectangle with width: $width, height: $length, color: $color, area: ${getArea()}")
+    }
+}
+
+class RightTriangle(val firstLeg: Double, val secondLeg: Double, override var color: Color) : Figure {
+    override fun getArea(): Double {
+        return 0.5 * firstLeg * secondLeg
+    }
+
+    override fun draw() {
+        println("Right Triangle with base: $firstLeg, height: secondLeg, color: $color, area: ${getArea()}")
```

Comment:
There is a typo in the `draw` method of the `RightTriangle` class. The `secondLeg` should be referenced correctly in the string interpolation. It should be `$secondLeg` instead of `secondLeg`.